docs/miscellaneous.rst

```diff
@@ -37,15 +37,18 @@ The elements of structs and arrays are stored after each other, just as if they
 Due to their unpredictable size, mapping and dynamically-sized array types use a Keccak-256 hash
 computation to find the starting position of the value or the array data. These starting positions are always full stack slots.
 
+Mapping and dynamic array
+=========================
+
 The mapping or the dynamic array itself
 occupies an (unfilled) slot in storage at some position ``p`` according to the above rule (or by
-recursively applying this rule for mappings to mappings or arrays of arrays). For a dynamic array, this slot stores the number of elements in the array (byte arrays and strings are an exception here, see below). For a mapping, the slot is unused (but it is needed so that two equal mappings after each other will use a different hash distribution).
-Array data is located at ``keccak256(p)`` and the value corresponding to a mapping key
+recursively applying this rule for mappings to mappings or arrays of arrays). For a dynamic array,
```

Comment:
I may be wrong, but to match 'arrays of arrays', should it be 'mappings of mappings'?